packages/core/src/tracing/instrumentation/process/process.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+const process = require('process');
+
+const { ENTRY } = require('../../constants');
+const shimmer = require('shimmer');
+const cls = require('../../cls');
+const spanBuffer = require('../../spanBuffer');
+
+const bullSpanName = require('../messaging/bull').spanName;
+
+let isActive = false;
+
+const bullKeepAlive = 10000;
+
+let currentlyActiveBullEntrySpan;
+
+exports.init = function init(config) {
+  if (config.tracing.activateBullProcessInstrumentation) {
+    shimmer.wrap(process, 'emit', shimProcessEmitForBullChildWorker);
+    shimmer.wrap(process, 'send', shimProcessSendForBullChildWorker);
+  }
+};
+
+function shimProcessEmitForBullChildWorker(originalProcessEmit) {
+  return function(event) {
+    if (!isActive || event !== 'message') {
+      return originalProcessEmit.apply(this, arguments);
+    }
+    const ipcMessage = arguments[1];
+
+    if (
+      //
+      !ipcMessage ||
+      ipcMessage.cmd !== 'start' ||
+      !ipcMessage.job ||
+      !ipcMessage.job.opts
+    ) {
+      return originalProcessEmit.apply(this, arguments);
+    }
+
+    const traceContext = ipcMessage.job.opts.instanaTracingContext;
+    delete ipcMessage.job.opts.instanaTracingContext;
+
+    if (
+      !traceContext ||
+      !traceContext.X_INSTANA_T ||
+      !traceContext.X_INSTANA_S ||
+      ipcMessage.job.opts.X_INSTANA_L === '0'
+    ) {
+      return originalProcessEmit.apply(this, arguments);
```

Comment:
Wouldn't we need to also do `delete ipcMessage.job.opts.X_INSTANA_L` here before calling `originalProcessEmit.apply`? OTOH, there are explicit tests for that I believe, so maybe this is handled somewhere else?

Reply:
That's a good catch. At the end, this check was not necessary, because the suppression is always handled in `bull.js#instrumentedProcessJob`.
In that function, `X_INSTANA_L` is already removed, and `X_INSTANA_S` and `X_INSTANA_T` are not even added to the `opts` object, which make the check of their existence enough already.